pkg/server/server.go

```diff
@@ -238,6 +290,16 @@ func (h *handler) authenticateEndpoint(w http.ResponseWriter, req *http.Request)
 	} else if userMapping, exists := h.lowercaseUserMap[arnLower]; exists {
 		username = userMapping.Username
 		groups = userMapping.Groups
+	} else if nodeMapping, exists := h.lowercaseNodeMap[arnLower]; exists {
+		privateDNS, err := h.getPrivateDNSFromEC2(identity.SessionName)
```

Comment:
either the function should return privateDNS or the function name should be getUsernameForNode

Reply:
Fixed.